src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -548,8 +549,8 @@ private string EncodeMimeWithQuotes(StringSegment input)
             var requiredLength = MimePrefix.Length +
                 Base64.GetMaxEncodedToUtf8Length(Encoding.UTF8.GetByteCount(input.AsSpan())) +
                 MimeSuffix.Length;
-            Span<byte> buffer = requiredLength <= 256
-                ? (stackalloc byte[256]).Slice(0, requiredLength)
+            Span<byte> buffer = requiredLength <= MaxStackSize
```

Comment:
👍🏻 
Additionally (not related to your PR per se): you can add `[SkipLocalsInit]` to this method (at L547) -- it's safe here as all content of `buffer` is written by this method, so it doesn't matter if zeroed or not.